src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -176,10 +128,11 @@ auto get_qemu_machine_type()
         return QString();
     }
 
-    auto process = set_qemu_exec();
-    process->setArguments({"-nographic", "-dump-vmstate", dump_file.fileName()});
-    process->start();
-    process->waitForFinished();
+    QProcess process;
```

Comment:
Would this not benefit from the Process treatment?

Reply:
Original goal was for Process to only wrap processes that need security confinement.

In this particular case, I don't think qemu here needs to be confined, as we ship it, and it isn't acting on any user-supplied data/file/arguments (I straced it to be sure, afaics it's just dumping internal data).

However I see demand for something more suitable to our needs than QProcess. Process is a candidate. Perhaps make it constructable, so it can be used just like QProcess too? Then rename ProcessFactory to ConfinedProcessFactory, to make clear the distinction in creation? WDYT?

Reply:
Yeah I think we might want to generalize Process a bit, but it can come later.